py/runtime.c

```diff
@@ -597,6 +597,9 @@ mp_obj_t MICROPY_WRAP_MP_BINARY_OP(mp_binary_op)(mp_binary_op_t op, mp_obj_t lhs
     if (type->binary_op != NULL) {
         mp_obj_t result = type->binary_op(op, lhs, rhs);
         if (result != MP_OBJ_NULL) {
+            if (op == MP_BINARY_OP_CONTAINS) {
+                return mp_obj_new_bool(mp_obj_is_true(result));
+            }
```

Comment:
it might be better to move this bit of code to `py/objtype.c:instance_binary_op` because that's really the only place it's needed, and it won't impact performance of other binary ops

Reply:
Moved it and removed the emit changes.